src/index.js

```diff
@@ -297,14 +300,20 @@ Multiaddr.prototype.equals = function equals (addr) {
  * // {family: 'IPv4', address: '127.0.0.1', port: '4001'}
  */
 Multiaddr.prototype.nodeAddress = function nodeAddress () {
-  if (!this.isThinWaistAddress()) {
-    throw new Error('Multiaddr must be "thin waist" address for nodeAddress.')
-  }
-
   const codes = this.protoCodes()
+  const names = this.protoNames()
   const parts = this.toString().split('/').slice(1)
+
+  if (parts.length < 4) {
+    throw new Error('Multiaddr must have a valid format: "{ip4, ip6, dns4, dns6}/{address}/{tcp, udp}/{port}".')
```

Comment:
Last mile! Typo `Multiaddr` should be `multiaddr`

Reply:
Oh, and valid format for a multiaddr is to start with `/` ;)

Reply:
Fixed :)